cmd/mavenStaticCodeChecks_test.go

```diff
@@ -0,0 +1,93 @@
+package cmd
+
+import (
+	"os"
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+
+	"github.com/SAP/jenkins-library/pkg/maven"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRunMavenStaticCodeChecks(t *testing.T) {
+	t.Run("should run spotBugs and pmd with all configured options", func(t *testing.T) {
+		execMockRunner := mock.ExecMockRunner{}
+		config := mavenStaticCodeChecksOptions{
+			PmdExcludes:               []string{"*test.java", "*prod.java"},
+			PmdRuleSets:               []string{"myRule.xml", "anotherRule.xml"},
+			SpotBugsExcludeFilterFile: "excludeFilter.xml",
+			SpotBugsIncludeFilterFile: "includeFilter.xml",
+			MavenModulesExcludes:      []string{"testing-lib", "test-helpers"},
+		}
+		expected := mock.ExecCall{
+			Exec: "mvn",
+			Params: []string{"-pl", "!unit-tests", "-pl", "!integration-tests",
+				"-pl", "!testing-lib", "-pl", "!test-helpers",
+				"-Dspotbugs.excludeFilterFile=includeFilter.xml",
```

Comment:
You may have encoded the bug in the test expectations. :-D

Reply:
copy paste -,- 
sorry